### PR TITLE
feat: move towards individual folder manifests

### PR DESCRIPTION
**Pull request by afonsojramos**
Welcome to PR number #766, this will be a big one, so bear with me.

This is something that I've thought about for quite a while and I think it is time that we move towards a centralised manifest point of view. That is, having a manifest in spicetify-themes that holds a readme with images to all themes + themes manifest. We then generate the "master" manifest with all the themes that are present. Think of it as the official DB for Marketplace.

In the end, we will have more granular control of what we display to end-users. And developers will continue to have all of the freedom to develop their themes and extensions within their repositories. In the end, the only change will be that the manifests that are present in the developers repo's will move to this repo.

This PR mostly attempts to fix https://github.com/spicetify/spicetify-marketplace/issues/111, however, it also ends up fixing https://github.com/spicetify/spicetify-marketplace/issues/137.

Once this PR is ready for primetime we will have a period where theme developers will be able to create PRs so that when we make the switch from `github topics` to `master manifest` every theme you love is present!

Additionally, I've added an image compression action so that the images we upload here do not occupy as much space.

### CODE REVIEW

**Review by theRealPadster — APPROVED** [review]

File: Dreary/manifest.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "Dreary",
+    "description": "Keep things boxed in and nice looking",
+    "preview": "Dreary/deeper.png",
+    "readme": "README.md",
```
Comment (theRealPadster) [inline Dreary/manifest.json:5]:
```suggestion
    "readme": "Dreary/README.md",
```

File: Glaze/manifest.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "Glaze",
+    "description": "Glaze",
+    "preview": "screenshots/dark.png",
+    "readme": "README.md",
```
Comment (theRealPadster) [inline Glaze/manifest.json:5]:
```suggestion
    "readme": "Glaze/README.md",
```

File: Glaze/manifest.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "Glaze",
+    "description": "Glaze",
+    "preview": "screenshots/dark.png",
```
Comment (theRealPadster) [inline Glaze/manifest.json:4]:
```suggestion
    "preview": "Glaze/screenshots/dark.png",
```

File: .github/workflows/compress-images.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        uses: actions/checkout@v2
+
+      - name: Compress Images
+        uses: calibreapp/image-actions@main
+        with:
+          jpegQuality: '70'
+          jpegProgressive: false
+          pngQuality: '70'
+          webpQuality: '70'
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
```
Comment (theRealPadster) [inline .github/workflows/compress-images.yml:28]: Maybe bump these to like 85 or something? Don't want to crush them

**Review by theRealPadster — CHANGES_REQUESTED** [review]

File: .github/workflows/compress-images.yml (hunk 4 above)
Comment (afonsojramos) [reply .github/workflows/compress-images.yml:28]: Checkout the diffs! 70 is not too low at all 

https://github.com/spicetify/spicetify-themes/pull/766/commits/f711cb314f22ad51c4928cab43a1534aa9ae1b33#diff-3c4454a0af8854a1ece53111c2fb0d25

Comment (theRealPadster) [reply .github/workflows/compress-images.yml:28]: Ah, nice

**Review by theRealPadster — APPROVED** [review]

**Comment by afonsojramos** [comment]
image

I also propose these changes to the manifest, since the README will always be the one in this repo, so it is redundant, and the `repository` field an optional one when the development is hosted outside of spicetify-themes.

**Review by theRealPadster — APPROVED** [review]

File: .github/workflows/manifest-generator.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+          await $`echo ${JSON.stringify(final_manifest, null, 2)} > manifest.json`
+          EOF
+
+      - name: push generated manifest
+        uses: EndBug/add-and-commit@v9
+        with:
+          add: 'manifest.json'
+          message: 'chore: update manifest.json'
+          committer_name: GitHub Actions
+          committer_email: 41898282+github-actions[bot]@users.noreply.github.com
```
[the remark this answers is not in the document]
Comment (afonsojramos) [reply .github/workflows/manifest-generator.yml:48]: That outputs a different icon, but it's whatever really 🤷‍♂️

File: .github/workflows/compress-images.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+
+      - name: Compress Images
+        uses: calibreapp/image-actions@main
+        with:
+          jpegQuality: '70'
+          jpegProgressive: false
+          pngQuality: '70'
+          webpQuality: '70'
```
[the remark this answers is not in the document]
Comment (afonsojramos) [reply .github/workflows/compress-images.yml:27]: Why not 30% less? The difference is truly negligible.

**Comment by CharlieS1103** [comment]
@afonsojramos A bit unrelated(but mostly related), however, how has progress been going on the spicetify-extensions repo? I am aware that you've just gotten back from vacation but I'm just wondering if it's something that is on your radar.

We have the functionality for the monomanifests all worked out on the marketplace side, all that's left now is to get the spicetify-extensions repo up and running, some finishing touches, and then merging it all together.

**Comment by theRealPadster** [comment]
(and one for apps, since we have a custom apps tab now as well)

**Comment by afonsojramos** [comment]
Expect it this weekend!! ✌️ I'm on a work trip, but I'll make sure to book some time to do it.

**Comment by afonsojramos** [comment]
@kyrie25 although I appreciate you trying to help, I'd like to understand your changes. Why did you manually edit `manifest.json`, and why did you remove Bloom? Was it because of the conflicts?

**Comment by kyrie25** [comment]
> @kyrie25 although I appreciate you trying to help, I'd like to understand your changes. Why did you manually edit `manifest.json`, and why did you remove Bloom? Was it because of the conflicts?

Yes, it was because of the conflicts as Bloom itself was removed quite some time in `main` itself

> Also, the invalid character started after some of your changes, why the need for all of these commits?

The invalid character appeared on my very first commit to this branch of what appears to be a CI problem where it'd remove the whitespaces from the `echo` command and the following was me trying to figure out why. You can check the CI run on my first commit to this branch to see what I was addressing to.

**Comment by harbassan** [comment]
Still happening?

**Review by Overrevvv — APPROVED** [review]

**Comment by CharlieS1103** [comment]
@Overrevvv I don't know why you're approving this, it's not even close to ready yet?

**Pull request closed without merging**